console/bridge.go

```diff
@@ -147,9 +148,15 @@ func (b *bridge) OpenWallet(call otto.FunctionCall) (response otto.Value) {
 		if input, err := b.prompter.PromptPassword("Please enter current PUK: "); err != nil {
 			throwJSException(err.Error())
 		} else {
+			if !regexp.MustCompile(`^[0-9]{12,}$`).MatchString(input) {
+				throwJSException("PUK needs to be at least 12 digit long")
+			}
 			pukpin = input
 		}
 		if input, err := b.prompter.PromptPassword("Please enter new PIN: "); err != nil {
+			if !regexp.MustCompile(`^[0-9]{6,}$`).MatchString(input) {
+				throwJSException("PIN needs to be at least 6 digit long")
+			}
```

Comment:
This should be in the else branch, not the error branch.

Reply:
ouch, yes